web-common/src/components/button-toggle-group/README.md

```diff
@@ -0,0 +1,58 @@
+the `<ButtonToggleGroup>` components allows the creation of radio button style controls that are displayed as a button group. this component maintains a bit of internal state to manage the selection of buttons, and dispatches events when those selections change. The main `<ButtonToggleGroup>` component must have one or more `<SubButton>` children, and no other elements are allowed. 
+
+Example:
+
+```javascript
+<ButtonToggleGroup>
+  <SubButton key={1}>
+    <Delta />%
+  </SubButton>
+  <SubButton key={2}>
+    <PieChart />%
+  </SubButton>
+</ButtonToggleGroup>
+```
+
+This implementation is based on a pattern recommended by Rich Harris here:
+https://stackoverflow.com/questions/56808584/iterate-over-slots-children-in-svelte-3
+
+https://svelte.dev/repl/8e68120858e5322272dc9136c4bb79cc?version=3.5.1
+
+
+---
+
+`<ButtonToggleGroup>` exposes the following props:
+
+`selectionRequired: bool = false` -- If this is `true`, then one button will always be selected, like a standard radio button. if `false`, then it is possible to untoggle all sub buttons. In either case, a maximum of 1 sub button may be selected
+
+`defaultKey: number | string` -- The key of the default selection.If undefined, the first sub button will be selected by default.
+
+`disabledKeys: (number | string)[] = [];` -- An array containing the keys of any sub buttons that are disabled.
+
+
+`<ButtonToggleGroup>` Dispatches the following events:
+
+"select-subbutton" -- When a sub button is selected, this event is dispatched with the key of the sub button
+
+"deselect-subbutton" -- When a sub button is deselected this event is dispatched with the key of the sub button. Note that this event may be fired by itself if no selection is required, indicating that the only selected button has been untoggled. If another sub button has been selected, then this event will be dispatched immediately before the selection event.
```

Comment:
See my previous comment; its unclear to me why you need separate events for these as opposed to just an event indicating the latest selected value